src/interactive/accordionPanelGroup.test.jsx

```diff
@@ -113,30 +116,28 @@ describe('AccordionPanelGroup', () => {
 			expect(closedPanels.length).toEqual(2);
 		});
 
-		it('changes panel open state with setPanelStates', () => {
+		it('changes panel open state with handlePanelClick', () => {
 			const panels = accordionPanelGroup.find(AccordionPanel);
-			const clickId = panels.at(0).prop('clickId');
+			const panelIndex = panels.at(0).prop('panelIndex');
 			const isOpen = panels.at(0).prop('isOpen');
 
-			expect(accordionPanelGroup.state('panelStates')[clickId]).toEqual(isOpen);
-			accordionPanelGroup.instance().setPanelStates(clickId, !isOpen);
-			expect(accordionPanelGroup.state('panelStates')[clickId]).toEqual(
+			expect(accordionPanelGroup.state('panelStatesList')[panelIndex]).toEqual(
+				isOpen
+			);
+			accordionPanelGroup
+				.instance()
+				.handlePanelClick({}, { panelIndex: panelIndex, isOpen: !isOpen });
+			expect(accordionPanelGroup.state('panelStatesList')[panelIndex]).toEqual(
 				!isOpen
 			);
 		});
 
-		it('calls componentWillReceiveProps on AccordionPanel primitive prop changes', () => {
-			const spy = jest.spyOn(
-				AccordionPanelGroup.prototype,
-				'componentWillReceiveProps'
-			);
-
-			const component = shallow(
+		it('calls getDerivedStateFromProps on AccordionPanel prop changes', () => {
```

Comment:
React itself has tests to ensure this lifecycle method is called as expected, and this test is indirectly testing `getNewPanelState` anyway. We could simplify this if...

The `getNewPanelState` tests have a few more assertions to check output based on different arguments, since this is what we're calling in `getDerivedStateFromProps` anyway. Those tests would cover expected state updates to the panel state list.

The tests around `getDerivedStateFromProps` here could then be simplified to just use a spy on `getNewPanelState` to verify it gets called after a `component.setProps()`. No need to assert expected state of a panel in here, as that would be covered by `getNewPanelState` tests.